### PR TITLE
`ultralytics 8.4.38` Unify args naming for standalone export functions

### PR DESCRIPTION
**Pull request by Laughing-q**
## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
🛠️ This PR standardizes export file and folder naming across multiple deployment formats, improves export API consistency, and fixes a CoreML input-name compatibility issue.

### 📊 Key Changes
- Unified export helpers to use clearer, explicit arguments like `output_file`, `output_dir`, `model`, and `im` instead of relying on source model filenames.
- Standardized exported artifact names across formats:
  - ExecuTorch now outputs `model.pte`
  - IMX now uses `model_imx.onnx`, `model_imx.pbtxt`, and `model_imx_MemoryReport.json`
  - OpenVINO now saves as `model.xml`
  - Other exporters now consistently write into named output directories such as `_executorch_model`, `_imx_model`, `_ncnn_model`, `_paddle_model`, `_rknn_model`, and `_axelera_model`
- Updated exporter functions to return output paths more consistently, making downstream handling simpler and more predictable.
- Refactored many export backends including TorchScript, OpenVINO, PaddlePaddle, MNN, NCNN, TensorFlow GraphDef, TensorFlow.js, Edge TPU, RKNN, Axelera, ExecuTorch, and IMX for cleaner interfaces.
- Fixed CoreML inference to use the actual model input name from the CoreML spec instead of assuming it is always `"image"` 🧠
- Improved OpenVINO export flexibility by supporting multiple example inputs and cleaning up import structure.
- Updated tests for ExecuTorch export to validate the new fixed filename `model.pte`.
- Refreshed Sony IMX500 docs to match the new exported filenames 📘
- Bumped Ultralytics package version from `8.4.37` to `8.4.38`.

### 🎯 Purpose & Impact
- Makes exported models easier to find and integrate because output names are now predictable across platforms ✅
- Reduces confusion for users deploying YOLO models to edge and hardware-specific runtimes, especially when switching between export targets.
- Improves maintainability for developers by giving export utilities a more consistent API and clearer parameter naming 🔧
- Prevents CoreML runtime failures in cases where the input tensor name is not `"image"`, improving compatibility on Apple devices 🍎
- Keeps documentation and tests aligned with actual exporter behavior, lowering the chance of deployment mistakes or broken assumptions.
- Overall, this is mainly a reliability and usability update rather than a model-quality change: exports should behave more consistently, but inference accuracy is not directly affected.

### CODE REVIEW

**Comment by Laughing-q** [comment]
@raimbekovm can you test the coreml changes here? thank you!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Laughing-q, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This is an automated message to help with review readiness, and an engineer will assist you shortly. Please review the checklist below to help ensure smooth integration. 😊

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Overall this is a clean, well-scoped export reliability PR. The CoreML input-name fix and the path normalization updates look good, but I found two issues worth addressing before merge: ONNX multi-input export should normalize lists to tuples, and TorchScript still has inconsistent path normalization when the input already has a `.torchscript` suffix.

💬 Posted 2 inline comments

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review 2

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Nice small reliability-focused PR overall, but there are two regressions to address: the OpenVINO single-input shape handling now changes semantics for the common one-input case, and the TorchScript path normalization no longer guarantees a string path for already-suffixed `Path` inputs.

💬 Posted 2 inline comments

**Comment by raimbekovm** [comment]
tested the coreml changes — export + autobackend inference works with different input names ("input", "image", custom), ImageType, multiple architectures (resnet, efficientnet, mobilenet, vit), and FP16. yolo11n regression test passes too. looks good to me

**Comment by Laughing-q** [comment]
## ⚡ Actions Trigger

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

GitHub Actions below triggered via workflow dispatch for this PR at 2026-04-03 10:55:37 UTC with `@ultralytics/run-ci` command:

* ✅ [CI](https://github.com/ultralytics/ultralytics/actions/runs/23943817710): `ci.yml` run 19980
* ✅ [Publish Docker Images](https://github.com/ultralytics/ultralytics/actions/runs/23943818093): `docker.yml` run 2031

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review 3

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

The arg-name cleanup is mostly solid, and the CoreML input-name fix looks like a good reliability improvement. I found a few path-handling issues in the new standalone export flow: Axelera now writes to CWD from the exporter, its cleanup can delete unrelated files in reused directories, and NCNN doesn't create parent directories for caller-supplied output paths.

💬 Posted 3 inline comments

**Comment by Laughing-q** [comment]
@raimbekovm I also unified the args naming for consistency for every export function.
- returning `str` type, expect openvino and some tensorflow related formats since `Exporter` currently needs them to return intermediate model so can have some additional integration on it.
- unified the args naming:
  - if source is torch model it's `model` and for onnx it's `onnx_file`.
  - if output is a file then it's `output_file` and for directory it's `output_dir`

**Comment by raimbekovm** [comment]
@Laughing-q tested with the new args naming, everything works well now

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review 4

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

The refactor is mostly mechanical and looks consistent, but I found two path/artifact regressions worth fixing before merge: Axelera now ignores the requested export directory, and OpenVINO now renames the saved XML to `model.xml`, which is likely to break downstream consumers.

💬 Posted 2 inline comments

File: ultralytics/utils/export/openvino.py [hunk 1]
```diff
@@ -53,10 +53,9 @@ def torch2openvino(
...
-    if file is not None:
-        file = Path(file)
-        suffix = f"_{'int8_' if int8 else ''}openvino_model{os.sep}"
-        f = str(file).replace(file.suffix, suffix)
-        f_ov = str(Path(f) / file.with_suffix(".xml").name)
-        ov.save_model(ov_model, f_ov, compress_to_fp16=half)
+    if output_dir is not None:
+        output_dir = Path(output_dir)
+        output_dir.mkdir(parents=True, exist_ok=True)
+        output_file = output_dir / "model.xml"
```
Comment (UltralyticsAssistant) [inline ultralytics/utils/export/openvino.py:59]: ⚠️ **HIGH**: This changes the saved IR filename from the model stem (for example `yolo11n.xml`) to a hardcoded `model.xml`. That is a breaking artifact-name change that isn't reflected elsewhere in this PR, and it can break downstream code that resolves the exported OpenVINO XML by the original basename. Please preserve the previous basename or update all consumers/tests in the same change.

File: ultralytics/engine/exporter.py [hunk 2]
```diff
@@ -1016,9 +1024,10 @@ def export_axelera(self, prefix=colorstr("Axelera:")):
 
         from ultralytics.utils.export.axelera import torch2axelera
 
+        output_dir = Path(f"{self.file.stem}_axelera_model")
```
Comment (UltralyticsAssistant) [inline ultralytics/engine/exporter.py]: 💡 **MEDIUM**: `self.file.stem` drops the configured parent directory, so Axelera exports will now be created in the current working directory instead of alongside the rest of the export artifacts. Since `torch2axelera()` no longer derives its own path from `self.file`, this becomes user-visible when `project`/`name` points somewhere else.

**Suggested change:**
```suggestion
        output_dir = self.file.parent / f"{self.file.stem}_axelera_model"
```

**Review by onuralpszr — APPROVED** [review]
LGTM !!

**Comment by onuralpszr** [comment]
## ⚡ Actions Trigger

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

GitHub Actions below triggered via workflow dispatch for this PR at 2026-04-09 09:40:48 UTC with `@ultralytics/run-ci` command:

* ✅ [CI](https://github.com/ultralytics/ultralytics/actions/runs/24183422232): `ci.yml` run 20181
* ✅ [Publish Docker Images](https://github.com/ultralytics/ultralytics/actions/runs/24183423009): `docker.yml` run 2048

**Comment by glenn-jocher** [comment]
## ⚡ Actions Trigger

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

GitHub Actions below triggered via workflow dispatch for this PR at 2026-04-10 12:16:14 UTC with `@ultralytics/run-ci` command:

* ✅ [CI](https://github.com/ultralytics/ultralytics/actions/runs/24242406423): `ci.yml` run 20305
* ✅ [Publish Docker Images](https://github.com/ultralytics/ultralytics/actions/runs/24242406965): `docker.yml` run 2060

**Comment by onuralpszr** [comment]
## ⚡ Actions Trigger

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

GitHub Actions below triggered via workflow dispatch for this PR at 2026-04-13 10:05:59 UTC with `@ultralytics/run-ci` command:

* ✅ [CI](https://github.com/ultralytics/ultralytics/actions/runs/24337608110): `ci.yml` run 20406
* ✅ [Publish Docker Images](https://github.com/ultralytics/ultralytics/actions/runs/24337608862): `docker.yml` run 2070

**Review by lakshanthad — APPROVED** [review]
RKNN, IMX and Axelera work as expected!

**Comment by onuralpszr** [comment]
## ⚡ Actions Trigger

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

GitHub Actions below triggered via workflow dispatch for this PR at 2026-04-15 13:22:26 UTC with `@ultralytics/run-ci` command:

* ✅ [CI](https://github.com/ultralytics/ultralytics/actions/runs/24456916289): `ci.yml` run 20531
* ✅ [Publish Docker Images](https://github.com/ultralytics/ultralytics/actions/runs/24456917202): `docker.yml` run 2079

**Comment by UltralyticsAssistant** [comment]
🎉 Huge congratulations on merging this excellent PR, @Laughing-q — and big thanks as well to @raimbekovm, @onuralpszr, @lakshanthad, and @glenn-jocher for the collaboration and review support.

> “Perfection is achieved, not when there is nothing more to add, but when there is nothing left to take away.” — Antoine de Saint-Exupéry

This update reflects that spirit beautifully: cleaner export APIs, predictable artifact naming across backends, and a thoughtful CoreML compatibility fix that makes deployment smoother and more reliable for everyone using Ultralytics. These kinds of consistency and usability improvements may look subtle on the surface, but they make a real difference for developers shipping YOLO models in production.

Thank you all for helping make exports more intuitive, maintainable, and dependable for the community 🚀
